RFC/src/RFC-0201_TariScript.md

```diff
@@ -278,74 +278,88 @@ pub struct TransactionInput {
+private values \\( (v\_i \\, , \\, k\_i )\\) and private script key \\(k\_{Si}\\) to prove ownership of thereof, see 
+[Signature on Commitment values] by F. Zhang et. al. and [Commitment Signature] by G. Yu. It signs the script, the 
+script input, _public script key_ and the commitment:
 
 $$
-  s_{Si} = r_{Si} + \hash{ R_{Si} \cat \alpha_i \cat \input_i \cat h_i} k_{Si}
-$$ 
-
-### Consensus changes
-
-The Mimblewimble balance for blocks and transactions stays the same.
+\begin{aligned}
+ s_{Si} = (a_{Si}, b_{Si}, R_{Si} )
+ \end{aligned}
+ \tag{3}
+$$
+Where
+$$
+\begin{aligned}
+R_{Si} &= r_{Si_a} \cdot H + r_{Si_b} \cdot G \\\\
+a_{Si}  &= r_{Si_a} +  e(v_{i}) \\\\
+b_{Si} &= r_{Si_b} +  e(k_{Si}+k_i) \\\\
+e &= \hash{ R_{Si} \cat \alpha_i \cat \input_i \cat K_{Si} \cat C_i} \\\\
+\end{aligned}
+\tag{4}
+$$
 
-The range proof commits all the values comprising the transaction output. So now instead of verifying the range proof
-using the standard output commitment, \\( C_i \\), we use the modified commitment,
+This is verified by the following:
+ $$
+\begin{aligned}
+a_{Si} \cdot H + b_{Si} \cdot G = R_{Si} + (C_i+K_{Si})e
+ \end{aligned}
+ \tag{5}
+$$
 
-$$ \hat{C_i} = v_i\cdot H  + \bigl( k_i + \rpc_i \bigr) \cdot G  $$
+This signature ensures that only the owner can provide the input data to the TransactionInput and that the transaction's 
+_script offset_ \\(\so\\) cannot be indirectly changed by changing the script private key \\(k\_{Si}\\).
 
-We can then verify the range proof. If the range proof is valid, we know that the value v, is positive and that none of
-the values have been changed.
+This stops any attack that attempts to change the information inside of the challenge \\(e\\) where more than one 
+possible input is valid such as with a `NOP` script. If an attacker can change the \\(K\_{Si}\\) keys of the input then 
+he can take control of the \\(K\_{Oi}\\) as well, allowing the attacker to change the metadata of the UTXO including the 
+script.
 
-The new offset pubkey, \\(K_O \\),  and the script public key, \\(K_S \\), are combined to create a script offset,
-\\( \so \\).
+### Script Offset
 
-\\( \so\\) is calculated and verified as part of every block and transaction validation. This is calculated as follows:
+For every transaction an accompanying script offset \\( \so \\) needs to be provided. This is there to prove that every 
+public script key \\( K\_{Sj} \\) and every _public script offset key_ \\( K\_{Oi} \\) supplied with the UTXOs are the 
+correct ones. The sender will know and provide _script offset private keys_ \\(k_{Oi} \\) and script private keys 
+\\(k_{Si} \\); these are combined to create the _script offset_ \\( \so \\), which is calculated as follows:
 
 $$
-\so = \sum_i\mathrm{k_{Si}} - \sum_j(\mathrm{k_{Oj}  \HU_j}) \\; \text{for each input}, i,\\, \text{and each output}, j
+\begin{aligned}
+\so = \sum_j\mathrm{k_{Sj}} - \sum_i\mathrm{k_{Oi}} \\; \text{for each input}, j,\\, \text{and each output}, i
```

Comment:
I think that this now allows cut through, you could remove a cut-through transaction and it's gamma from the block and publish the new gamma